File: Java/src/NapakalakiGame/Treasure.java
/*
 * Autores:
 * Guillermo Montes Martos
 * Manuel Ruiz López
 *
 */
package NapakalakiGame;

public class Treasure {
    private String name;
    private int bonus;
    private TreasureKind type;
    

    // Constructor
       
    public Treasure (String n, int b, TreasureKind t){
        name = n;
        bonus = b;
        type = t;
    }
    
    public String getName(){
        return name;
    }
    
    public int getBonus(){
        return bonus;
    }
    
    public TreasureKind getType(){
        return type;
    }
    
    @Override
    public String toString() {
        return (name+" (Bonus: "+bonus+", Tipo: "+type+")");
    }
}

File: Ruby/lib/Treasure.rb
#encoding: utf-8

# Autores:
# 
#     Guillermo Montes Martos
#     Manuel Ruiz López
#

require_relative 'TreasureKind.rb'

module NapakalakiGame
class Treasure
  
  attr_reader :name
  attr_reader :bonus
  attr_reader :type
  
  def initialize(aName, aBonus, aType)
    @name = aName
    @bonus = aBonus
    @type = aType
  end
  
  def to_s()
    "#{@name} (#{@type}, +#{@bonus})"
  end
  
end
end